src/engine/idTable/IdTable.h

```diff
@@ -773,6 +774,31 @@ class IdTable {
         });
   }
 
+  // Add the rows with the specified `indices` from the `table` at the end of
+  // this IdTable. The order of the inserted rows is the same as in `indices`.
+  // The input must be some kind of `IdTable`.
+  template <typename Table>
+  void insertSubsetAtEnd(const Table& table,
+                         const std::vector<size_t>& indices) {
+    AD_CORRECTNESS_CHECK(table.numColumns() == numColumns());
+    const size_t numInserted = indices.size();
+    if (numInserted == 0) return;
+
+    const size_t srcSize = table.size();
+    AD_EXPENSIVE_CHECK(ql::ranges::all_of(
+        indices, [srcSize](size_t idx) { return idx < srcSize; }));
+    const size_t oldSize = size();
+    resize(numRows() + numInserted);
+    // For each column, copy the requested rows into the reserved tail.
+    for (auto&& [destination, source] :
+         ranges::views::zip(ad_utility::allView(getColumns()),
```

Comment:
```suggestion
         ::ranges::views::zip(ad_utility::allView(getColumns()),
```
(our canonical way to represent the `range-v3` library